File: path_sum.c
#include<stdio.h>
#include<stdlib.h>
#include<stdbool.h>

struct TreeNode {
    int val;
    struct TreeNode *left;
    struct TreeNode *right;
};

bool hasPathSum(struct TreeNode* root, int sum) {
    //int sum = 0;
    struct TreeNode *current = root;
    if(!current)
        return false;
    else if(current->val == sum && !current->left && !current->right)
        return true;
    else{
        return hasPathSum(current->left,sum-current->val) || hasPathSum(current->right,sum-current->val);
    }
}

File: py/tree/PathSum.py
# Definition for a binary tree node.
class TreeNode(object):
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution(object):
    def hasPathSum(self, root, sum):
        if root == None:
            return False
        if root.left == None and root.right == None and root.val == sum:
            return True
        return self.hasPathSum(root.left, sum-root.val) or self.hasPathSum(root.right, sum-root.val)
